kernel/init.c

```diff
@@ -188,7 +188,7 @@ void _data_copy(void)
 #endif	/* CONFIG_CODE_DATA_RELOCATION */
 #ifdef CONFIG_APP_SHARED_MEM
 	(void)memcpy(&_app_smem_start, &_app_smem_rom_start,
-		 ((u32_t) &_app_smem_end - (u32_t) &_app_smem_start));
```

Comment:
It's not clear to me that this optimizes anything, wouldn't the compiler do this arithmetic at build time anyway?

Reply:
GCC was doing something very strange, the smem_end and smem_start value changes with new objects getting added. But it was not getting reflected when the code was run. This was a strange situation. hence i had to use a  "_size"  liker variable to get this done.